txn_writer.go

```diff
@@ -418,38 +418,51 @@ fail:
 }
 
 // PrintPreparedActions logs the actions a call to Write would perform.
-func (sw *SafeWriter) PrintPreparedActions(output *log.Logger) error {
+func (sw *SafeWriter) PrintPreparedActions(verbose bool, output *log.Logger) error {
 	if sw.HasManifest() {
-		output.Printf("Would have written the following %s:\n", ManifestName)
-		m, err := sw.Manifest.MarshalTOML()
-		if err != nil {
-			return errors.Wrap(err, "ensure DryRun cannot serialize manifest")
+		if verbose {
+			m, err := sw.Manifest.MarshalTOML()
+			if err != nil {
+				return errors.Wrap(err, "ensure DryRun cannot serialize manifest")
+			}
+			output.Printf("Would have written the following %s:\n%s\n", ManifestName, string(m))
+		} else {
+			output.Printf("Would have written %s.\n", ManifestName)
 		}
-		output.Println(string(m))
 	}
 
 	if sw.writeLock {
 		if sw.lockDiff == nil {
-			output.Printf("Would have written the following %s:\n", LockName)
-			l, err := sw.lock.MarshalTOML()
-			if err != nil {
-				return errors.Wrap(err, "ensure DryRun cannot serialize lock")
+			if verbose {
+				l, err := sw.lock.MarshalTOML()
+				if err != nil {
+					return errors.Wrap(err, "ensure DryRun cannot serialize lock")
+				}
+				output.Printf("Would have written the following %s:\n%s\n", LockName, string(l))
+			} else {
+				output.Printf("Would have written %s.\n", LockName)
 			}
-			output.Println(string(l))
 		} else {
-			output.Printf("Would have written the following changes to %s:\n", LockName)
-			diff, err := formatLockDiff(*sw.lockDiff)
```

Comment:
I definitely think that the vendor info e.g. `github.com/stretchr/testify@v1.1.4 with packages: [assert]` should only be logged in verbose mode, since we don't have vendor verification and therefore every package is listed regardless of what's changed.

Not so sure about the manifest though. I think it will print the entire manifest contents, even when only 1 thing has changed, right? Basically we don't have manifest diffing? From a user prespective, I'd prefer to only see changes when I ask for a dry-run, and put anything that is a "mega dump" under the verbose flag.